### PR TITLE
dm-syncer: add config for task name

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
fix issue https://github.com/pingcap/dm/issues/478
if all dm-syncer use one task name, they will update same checkpoint table.

### What is changed and how it works?

add a config for task name, and it can't be empty

### Check List 

Tests 

 - Integration test

### CODE REVIEW

**Comment by GregoryIan** [comment]
@WangXiangUSTC Please fix CI

File: tests/dm_syncer/run.sh [hunk 1]
```diff
@@ -50,7 +50,7 @@ function run() {
     cat $cur/conf/dm-syncer-2.toml > $WORK_DIR/dm-syncer-2.toml
     cat $cur/conf/old_meta_file > $WORK_DIR/old_meta_file
     # $worker1_run_source_1 > 0 means source1 is operated to worker1
-    worker1_run_source_1=$(sed "s/$SOURCE_ID1/$SOURCE_ID1\n/g" $WORK_DIR/worker1/log/dm-worker.log | grep -c "$SOURCE_ID1")
+    worker1_run_source_1=$(sed "s/$SOURCE_ID1/$SOURCE_ID1\n/g" $WORK_DIR/worker1/log/dm-worker.log | grep -c "$SOURCE_ID1") || true
```
Comment (WangXiangUSTC) [inline tests/dm_syncer/run.sh]: if can't find `SOURCE_ID1` in  log, will exit with 1, so just add `|| true`

File: cmd/dm-syncer/config.go [hunk 2]
```diff
@@ -99,7 +102,7 @@ func (c *commonConfig) newConfigFromOldConfig(args []string) (*config.SubTaskCon
 	fs.IntVar(&cfg.MaxRetry, "max-retry", 100, "maxinum retry when network interruption")
 	fs.BoolVar(&cfg.EnableANSIQuotes, "enable-ansi-quotes", false, "enable ANSI_QUOTES sql_mode")
 	fs.StringVar(&cfg.TimezoneStr, "timezone", "", "target database timezone location string")
-	fs.BoolVar(&OldConfigFormat, "old-config-format", false, "read old config format")
+	fs.BoolVar(&SyncerConfigFormat, "syncer-config-format", false, "read dyncer config format")
```
Comment (csuzhangxc) [inline cmd/dm-syncer/config.go]:
```suggestion
	fs.BoolVar(&SyncerConfigFormat, "syncer-config-format", false, "read syncer config format")
```

File: cmd/dm-syncer/config.go [hunk 3]
```diff
@@ -302,14 +308,14 @@ func loadMetaFile(metaFile string) (*config.Meta, error) {
...
+func (oc *syncerConfig) convertToNewFormat() (*config.SubTaskConfig, error) {
 	meta, err := loadMetaFile(oc.Meta)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	newTask := &config.SubTaskConfig{
-		Name:     "dm-syncer",
-		SourceID: "dm-syncer-from-old-config",
+		Name:     oc.Name,
+		SourceID: fmt.Sprintf("%s:%d", oc.From.Host, oc.From.Port),
```
Comment (csuzhangxc) [inline cmd/dm-syncer/config.go]: If the user config `From` to another instance in the same replica group and enabled GTID mode (supported later), then the checkpoint can't be used anymore?

File: cmd/dm-syncer/config.go [hunk 4]
```diff
@@ -34,6 +34,8 @@ import (
 type commonConfig struct {
 	*flag.FlagSet `json:"-"`
 
+	// task name
+	Name         string
```
Comment (csuzhangxc) [inline cmd/dm-syncer/config.go]: Do we need to ensure it's not empty?

Comment (WangXiangUSTC) [reply cmd/dm-syncer/config.go]: this will be checked by `subtask config`

File: cmd/dm-syncer/config.go (hunk 3 above)
Comment (WangXiangUSTC) [reply cmd/dm-syncer/config.go]: update in 41f8c7a

**Pull request merged**